sanity_html/types.py

```diff
@@ -73,23 +73,27 @@ def get_node_siblings(self, node: Union[dict, Span]) -> Tuple[Optional[dict], Op
         if not self.children:
             return None, None
         try:
-            if not isinstance(node, (dict, Span)):
-                raise ValueError(f'Expected dict or Span but received {type(node)}')
-            elif type(node) == dict:
+            if type(node) == dict:
                 node = cast(dict, node)
                 node_idx = self.children.index(node)
             elif type(node) == Span:
                 node = cast(Span, node)
-                node_idx = self.children.index(next((c for c in self.children if c.get('_key') == node._key), {}))
+                for index, item in enumerate(self.children):
+                    if node.text == item['text']:
+                        # Is it possible to handle several identical texts?
+                        node_idx = index
+                        break
+            else:
+                raise ValueError(f'Expected dict or Span but received {type(node)}')
```

Comment:
Since we had no else-clause in the previous setup, there was an unhandled code path. Flake8 error SIM106 wanted us to put guard clauses at the start, but that would require us to have 

```diff
guard clause
elif
elif
+ additional else 
```

and that just seems excessive to me